Adds integration test for vis and modifies unit tests for vis

bhuvangupta008 opened the pull request:
## What? Why?
Fix #283  . (Add Reference to the issue if any)

Changes proposed in this pull request:
- a Refactored unit test for vis module(mocked some calls)
- b Added integration test for vis module

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?

## Notify reviewers
@kaushik-rohit
@prasadtalasila

bhuvangupta008 committed:
Adds integration test for vis and modifies unit tests for vis

bhuvangupta008 committed:
Adds integration test for vis and modifies unit tests for vis

bhuvangupta008 committed:
Modifies unit test for network and slack modules for patching issues #278 (#285)

prasadtalasila commented on test/unit-test/test_lib/test_vis.py:
```diff
@@ -1,49 +1,53 @@
-sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
-current_dir = os.path.dirname(os.path.abspath(__file__))
 
+def mock_exponential_curve_func(x, a, b, c):
```
Since these mock functions are being used only inside the test class in this module, is it better to move the functions into the VisTest class?
@Kausam and @kaushik-rohit please pitch in with the prevailing convention for this kind of mock functions.

prasadtalasila reviewed (changes requested):
@bhuvangupta008 please see the suggested changes.

Kausam committed:
add integration test for keywords function of user module (#284)

* adds integration test for lib.analysis.user.keywords()
* updates scripts to run unit test and integration test

kaushik-rohit committed:
network: Improve tests to increase coverage (#288)

1) ubunutu module
	a) increase log data to cover graph coloring in hits.

2) slack
	a) remove undirected graph part from degree_analysis_on_graph
	since we don't use it anywhere.

Issue Link: https://github.com/prasadtalasila/IRCLogParser/issues/258

bhuvangupta008 committed:
Merge branch 'dev' of git://github.com/prasadtalasila/IRCLogParser into vis-integration-test
